i18n/ja/docusaurus-plugin-content-docs/current/guides/integration-dsp-no-sdk.md

```diff
@@ -0,0 +1,126 @@
+---
+title: DSP Direct Integration Tips
+sidebar_label: DSP Direct Integration Tips
+description: Tips for DSPs who are integrating with UID2 without using SDKs.
+hide_table_of_contents: false
+sidebar_position: 05
+---
+
+import Link from '@docusaurus/Link';
+
+# DSP Direct Integration Tips
+
+This document provides integration tips for DSPs who prefer to integrate without using one of the UID2 SDKs.
+
+If you're a DSP and prefer to use an SDK, see [SDKs: Summary](../sdks/summary-sdks.md).
+
+## Overview
+
+DSPs must be able to decrypt UID2 tokens to raw UID2s and verify the token validity, so that they can use the decrypted tokens for targeted advertising and bidding purposes. To do this, a DSP must do the following:
+
+- [Download a set of encryption keys](#downloadrefresh-encryption-keys)
+- [Decrypt UID2 tokens into raw UID2s](#decrypt-uid2-tokens-into-raw-uid2s)
+- [Periodically re-download the latest set of encryption keys](#downloadrefresh-encryption-keys)
+
+An example implementation, in the [UID2 SDK for C#&nbsp;/&nbsp;.NET](https://github.com/IABTechLab/uid2-client-net), is the `BidstreamClient` class: see [BidstreamClient.cs](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/BidstreamClient.cs).
+
+This document refers to additional code sections from the C#&nbsp;/&nbsp;.NET SDK as examples.
+
+## Download/Refresh Encryption Keys
+
+To download or refresh the encryption keys, so that you can decrypt UID2 tokens into raw UID2s, call the following API endpoint:
+
+```
+/v2/key/bidstream
+```
+
+Refresh the keys once at startup, and then periodically (recommended refresh interval is hourly).
+
+The UID2 SDK for C#&nbsp;/&nbsp;.NET uses a `Refresh` function.
+
+To see how all the fields are parsed, refer to the UID2 SDK for C#&nbsp;/&nbsp;.NET `Parse` function: see [KeyParser.cs, lines 41-74](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/KeyParser.cs#L41-L74).
+
+The API response is in JSON format, and includes `site_data`, the list of domains or app names that are allowed for the site or app.
+
+After decrypting the token into a raw UID2, for client-side integrations, you can use the information in `site_data` to verify that a specific domain or app name is on the list of names allowed for tokens that are generated on the client side. For details, see [Verify the Domain or App Name](#verify-the-domain-or-app-name).
+
+To refresh the keys, you'll need to do the following:
```

Comment:
Hi Gen, I apologize for the lack of clarity in my previous comment. To call the refresh function to obtain the keys, it's not just about encrypting the request or decrypting the response. There are additional steps involved, as the doc mentioned above: calling the endpoint and parsing the response. What I want to emphasize here is that we also need to decrypt the request or encrypt the response and attach the link about how to do it.

Reply:
Ah got it @caroline-ttd thank you! I reworked that section. LMK any input. Thanks!